### PR TITLE
Be ca 0126

### PR DESCRIPTION
**Pull request by habibdevsy**
- closestorders
- acceptedorder
- getSpecificOrderForCaptain
- resolve (captain id) in (order) with new captain entity

### CODE REVIEW

**Review by ralsalloum — APPROVED** [review]

File: backend/src/Manager/Captain/CaptainManager.php [hunk 1]
```diff
@@ -149,4 +149,14 @@ public function checkCompleteAccountStatusValidity(string $completeAccountStatus
 
         return true;
     }
+    
+    public function captainIsActive($captainId): ?array
```
Comment (ralsalloum) [inline backend/src/Manager/Captain/CaptainManager.php:153]: type hint is missing for these two functions.

File: backend/src/Controller/Order/OrderController.php [hunk 2]
```diff
@@ -292,4 +293,170 @@ public function filterStoreOrders(Request $request): JsonResponse
...
+    public function acceptedOrderByCaptainId(): JsonResponse
+    {
+        $result = $this->orderService->acceptedOrderByCaptainId($this->getUserId());
+
+        return $this->response($result, self::FETCH);
+    }
+    
+    /** captain: Get order details.
+     * @Route("captainorder/{id}", name="getSpecificOrderForCaptain", methods={"GET"})
+     * @IsGranted("ROLE_OWNER")
```
Comment (ralsalloum) [inline backend/src/Controller/Order/OrderController.php:405]: Must be for Captain Not Store Owner privilege's

File: backend/src/Manager/Order/OrderManager.php [hunk 3]
```diff
@@ -100,4 +102,21 @@ public function updateOrderStateBySuperAdmin(OrderStateUpdateBySuperAdminRequest
             return $orderEntity;
         }
     }
+
+    public function closestOrders(): ?array
+    {
+        return $this->orderRepository->closestOrders();
+    }
+    
+    public function acceptedOrderByCaptainId($userId): ?array
```
Comment (ralsalloum) [inline backend/src/Manager/Order/OrderManager.php:111]: type hint is missing for the new functions.

File: backend/src/Repository/CaptainEntityRepository.php [hunk 4]
```diff
@@ -60,4 +60,18 @@ public function getCompleteAccountStatusByCaptainId($captainId): ?array
             ->getQuery()
             ->getOneOrNullResult();
     }
+    
+    public function captainIsActive($captainId): ?array
```
Comment (ralsalloum) [inline backend/src/Repository/CaptainEntityRepository.php:64]: type hint is missing.

File: backend/src/Repository/OrderEntityRepository.php [hunk 5]
```diff
@@ -130,4 +134,62 @@ public function filterStoreOrders(OrderFilterRequest $request, $storeOwner): ?ar
...
+           
+            ->andWhere('orderEntity.state = :pending ')
+
+            ->setParameter('pending', OrderStateConstant::ORDER_STATE_PENDING)
+
+            ->getQuery()
+            ->getResult();
+    }
+    
+    public function acceptedOrderByCaptainId($captainId): ?array
```
Comment (ralsalloum) [inline backend/src/Repository/OrderEntityRepository.php:152]: type hint is missing.

File: backend/src/Controller/Order/OrderController.php (hunk 2 above)
Comment (Kenan-Hussein) [reply backend/src/Controller/Order/OrderController.php:405]: @ralsalloum this is a danger bug please contact Habib for insurances

Comment (Kenan-Hussein) [reply backend/src/Controller/Order/OrderController.php:405]: seems fixed. thanks

**Pull request merged**